packages/flutter_tools/lib/src/version.dart

```diff
@@ -217,52 +217,20 @@ class FlutterVersion {
     }
   }
 
-  /// The name of the temporary git remote used to check for the latest
-  /// available Flutter framework version.
-  ///
-  /// In the absence of bugs and crashes a Flutter developer should never see
-  /// this remote appear in their `git remote` list, but also if it happens to
-  /// persist we do the proper clean-up for extra robustness.
-  static const String _versionCheckRemote = '__flutter_version_check__';
-
-  /// The date of the latest framework commit in the remote repository.
+  /// The latest commit hash of the remote repository.
   ///
   /// Throws [VersionCheckError] if a git command fails, for example, when the
   /// remote git repository is not reachable due to a network issue.
-  static Future<String> fetchRemoteFrameworkCommitDate(String branch) async {
-    await _removeVersionCheckRemoteIfExists();
+  static Future<String> fetchRemoteFrameworkLatestRevision() async {
+    String revision;
     try {
-      await _run(<String>[
```

Comment:
Why did you get rid of this temporary remote? This existed specifically so that we could check if the `_flutterGit` repo has upstream changes, not relying on whatever the user currently configured as their upstream.

Reply:
Yeah I'm also not sure about this. I figured since it did the same thing as the following lines in `upgrade.dart`:
https://github.com/flutter/flutter/blob/a17bbb9bbe7a44603c5b155416720ac22726fcf0/packages/flutter_tools/lib/src/commands/upgrade.dart#L222-L233
i.e, getting the head of the upstream, we could skip making the remote.
(wouldn't `upgrade.dart` have the same issue?)

As suggested, I won't change this in the feature PR. Let me know about your thoughts when I send the cleanup PR.

Reply:
So upgrade will use whatever is configured as the user's upstream (for many developers, this ends up being their personal fork of Flutter). However, the version check is meant to check either the official Flutter repo github.com/flutter/flutter, or an automated mirror of that if they set the env var `FLUTTER_GIT_URL`